# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: ipc/chromium/src/base/pickle.cc
# Commit: 451508713a62
# Full Hash: 451508713a625beaaa9fcbbf06c65ca312dc34fb
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2021-06-23 09:53:24
# Regressor Bug: 1706374
# File Overlap Count: 2
# Description:
#   Bug 1706374 - Part 13: Remove the event footer from an IPC::Message when deserializing, r=handyman,glandium
#   
#   This unfortunately requires a new method to be added to BufferList to
#   support truncating the buffer to a particular iterator.
#   
# ==============================================================================

diff -r 878722e87e62 -r 451508713a62 ipc/chromium/src/base/pickle.cc
--- a/ipc/chromium/src/base/pickle.cc	Tue Jun 22 18:17:23 2021 +0000
+++ b/ipc/chromium/src/base/pickle.cc	Tue Jun 22 18:17:23 2021 +0000
@@ -471,6 +471,11 @@
   }
 }
 
+void Pickle::Truncate(PickleIterator* iter) {
+  size_t dropped = buffers_.Truncate(iter->iter_);
+  header_->payload_size -= dropped;
+}
+
 void Pickle::BeginWrite(uint32_t length, uint32_t alignment) {
   DCHECK(alignment % 4 == 0) << "Must be at least 32-bit aligned!";
 